Create a solutions page for Mattermost Customizable ChatGPT Bot Framework (#6355)
mattermost/docs: source/about/mattermost-customizable-chatgpt-bot-framework.rst

.. _mattermost-customizable-chatgpt-bot-framework:

> Mattermost Customizable ChatGPT Bot Framework

Reviewer: Good clarification.

> Mattermost Customizable ChatGPT Bot Framework

Author: @crspeller Thanks for the feedback!

[On our other thread](https://docs.google.com/document/d/1Y9Z3kfzwP07IsqO8VjAAmetiaMg6flGrFLcVgc3EgOc/edit?disco=AAAAvRhIfbQ) we talked a doc page for "Mattermost customizable ChatGPT bot framework" with a vision of generative AI bot integrations, including links to examples and tradeoffs (e.g. one being unavailable for commercial use). 

Would there be a way to adjust the title and/or introduction to this plan?

> Mattermost Customizable ChatGPT Bot Framework

Reviewer 2: ```suggestion
Mattermost customizable ChatGPT bot framework
```

> Mattermost Customizable ChatGPT Bot Framework

Reviewer 3: Speak in future tense or change the title. It sounds in some places we are describing something that exists. If this is aspirational it should be clear what we have today (the plugins) and what we might build (the framework).

========================================================

> Mattermost is committed to providing an open source platform for secure collaboration and control, especially in strict security environments. With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions.

Reviewer: Proposing:

"Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions **without sacrificing security**."

I think driving the security point again at the end strengthens the proposed copy.

With a focus on AI augmentation and integration, Mattermost is providing an open source customizable ChatGPT bot framework with full control and data privacy for organizations that want to deploy AI solutions without sacrificing security.

> Control and Data Privacy

Reviewer 2: ```suggestion
Control and data privacy
```

> Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data and the ability to use private cloud AI solutions like Azure AI or AWS.

Reviewer: Adding links will help open the door for brainstorming and can help them envision their AI on a Mattermost deployment.

Azure AI: https://azure.microsoft.com/en-us/solutions/ai/#benefits
AWS AI: https://aws.amazon.com/machine-learning/ai-services/

> Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data and the ability to use private cloud AI solutions like `Azure AI <https://azure.microsoft.com/en-us/solutions/ai/#benefits>`_ or `AWS <https://aws.amazon.com/machine-learning/ai-services/>`_.

Reviewer 2: ```suggestion
Mattermost's open source platform provides verifiable privacy assurances, data control, and granular permissions solutions. As AI becomes more prevalent globally, organizations need control over the AI's access to their data, and the ability to use private cloud AI solutions like `Azure AI <https://azure.microsoft.com/en-us/solutions/ai/#benefits>`__ or `AWS <https://aws.amazon.com/machine-learning/ai-services/>`__.
```

In the short term, this may involve integrating private-cloud AI with Mattermost, while in the long term, it could include training foundation models with an organization's data in a controlled environment.

> Get Started with the Sample Open Source ChatGPT Plugin

Reviewer 2: ```suggestion
Get started with the sample open source ChatGPT plugin
```

--------------------------------------------------------

> The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks, enabling increased speed and efficiency in government, defense, and technology organizations.

Reviewer 2: ```suggestion
The customizable ChatGPT bot framework allows users to integrate generative AI solutions like OpenAI's ChatGPT and private cloud LLMs (Large Language Models) into their Mattermost workflows. This integration can streamline communication, enhance decision-making, and automate tasks to enable increased speed and efficiency in government, defense, and technology organizations.
```

An example of the customizable ChatGPT bot framework is the `open source ChatGPT plugin for Mattermost <https://github.com/Brightscout/mattermost-plugin-openai>`_ developed by Brightscout. To get started, follow these steps:

1. Visit the `GitHub repository <https://github.com/Brightscout/mattermost-plugin-openai>`_ for the ChatGPT plugin.

> 2. Download the `latest release from the Releases <https://github.com/Brightscout/mattermost-plugin-openai/releases>`_ section.

Reviewer 2: ```suggestion
2. Download the `latest release <https://github.com/Brightscout/mattermost-plugin-openai/releases>`__ from the **Releases** section.
```

> 5. Create a bot account in Mattermost and copy the bot access token.

Reviewer: @jasonblais I don't believe steps 5 and 6 are necessary for this plugin. They're not listed anywhere on the repo. I think these steps were only needed for Sebastian Muller's plugin.

Reviewer 2: @jasonblais - Please note @azigler's feedback here.

> 3. Upload the release file to your Mattermost server via **System Console > Plugins > Plugin Management**.

Reviewer 2: ```suggestion
3. Upload the release file to your Mattermost server by going to **System Console > Plugins > Plugin Management**.
```

4. Enable the plugin and configure it using your OpenAI API key.

> Now, you can start using the ChatGPT plugin in your Mattermost instance.

Reviewer 3: ```suggestion
Now, you can start using the OpenAI plugin in your Mattermost instance.
```

Two other examples of ChatGPT integrations built by Sebastian Müller include:

> - `ChatGPT Mattermost Bot with OpenAI <https://github.com/yGuy/chatgpt-mattermost-bot>`_

Reviewer 2: ```suggestion
- `ChatGPT Mattermost Bot with OpenAI <https://github.com/yGuy/chatgpt-mattermost-bot>`__
```

> - `ChatGPT Mattermost Bot with LLaMa <https://github.com/yGuy/chatgpt-mattermost-bot/tree/llama>`_

Reviewer: I recommend adding a `*` at the end of this line and the beginning of the LLaMa disclaimer, to enforce the connection.

Reviewer 2: @jasonblais - I agree with @azigler's feedback here.
